src/crimson/osd/backfill_state.cc

```diff
@@ -493,6 +493,14 @@ BackfillState::Crashed::Crashed()
   ceph_abort_msg("{}: this should not happen");
 }
 
+// -- Cancelled
+BackfillState::Cancelled::Cancelled()
+{
+  backfill_state().backfill_info.clear();
+  backfill_state().peer_backfill_info.clear();
+  backfill_state().progress_tracker.reset();
```

Comment:
```
  if (!waiting_on_backfill.empty()) {
    waiting_on_backfill.clear();
   }
```

We are not canceling ongoing backfill operations here, right?

Reply:
> ~if (!waiting_on_backfill.empty()) {~ ~waiting_on_backfill.clear();~ ~}~
> 
> We are not canceling ongoing backfill operations here, right?

Yeah, we chose to let ongoing backfill operations finish but prevent the backfill to proceed any longer.

Reply:
Optimally, we should probably also cancel ongoing ones I believe. Although this can be a follow up PR.

Reply:
> Optimally, we should probably also cancel ongoing ones I believe. Although this can be a follow up PR.

Yeah, we might need to add more interrupt conditions in the future.

Reply:
Can you please open a low priority tracker and paste it here?